### PR TITLE
Fix bug which caused all per-entity data to be sent to a single partitioner before sampling

### PR DESCRIPTION
**Pull request by ashelkovnykov**
Results in crashes for cases which need the 'active.data.upper.bound'.

### CODE REVIEW

File: photon-api/src/main/scala/com/linkedin/photon/ml/data/RandomEffectDataset.scala [hunk 1]
```diff
@@ -406,29 +406,17 @@ object RandomEffectDataset {
...
-        val projectorLookupTable = projectorsIt.toMap
-
-        dataIt.map { case (rEID, (uID, LabeledPoint(label, features, offset, weight))) =>
-
-          val projector = projectorLookupTable(rEID)
-          val projectedFeatures = projector.projectForward(features)
-
-          (rEID, (uID, LabeledPoint(label, projectedFeatures, offset, weight)))
-        }
+      .join(projectors)
```
Comment (limkothari) [inline photon-api/src/main/scala/com/linkedin/photon/ml/data/RandomEffectDataset.scala]: Is it possible that the projectors are still causing the data to be keyed by partitions and sent to single executor to be loaded in the memory?

Comment (ashelkovnykov) [reply photon-api/src/main/scala/com/linkedin/photon/ml/data/RandomEffectDataset.scala]: I think I see what you mean - @yunboouyang and I just spoke and we're working on another solution. Yes, the same partitioner is being used even without being specified as the keyed data loses its partitioner, so Spark reuses the partitioner of the projector RDD.

Comment (limkothari) [reply photon-api/src/main/scala/com/linkedin/photon/ml/data/RandomEffectDataset.scala]: Thanks @ashelkovnykov ! Please let me know the patch and i can verify it with my dataset :)

File: photon-api/src/main/scala/com/linkedin/photon/ml/data/RandomEffectDataset.scala [hunk 2]
```diff
@@ -271,37 +271,38 @@ object RandomEffectDataset {
     //
 
     val keyedGameDataset = generateKeyedGameDataset(gameDataset, randomEffectDataConfiguration)
-    keyedGameDataset.persist(StorageLevel.MEMORY_ONLY_SER).count
+    keyedGameDataset.persist(StorageLevel.MEMORY_AND_DISK_SER).count
```
Comment (yunboouyang) [inline photon-api/src/main/scala/com/linkedin/photon/ml/data/RandomEffectDataset.scala]: Curious to know the reason of changing the storage level here

**Comment by yunboouyang** [comment]
Fixed the bug in https://github.com/linkedin/photon-ml/pull/466

**Pull request closed without merging**